feat: Add flag for legacy trust store option

esolitos opened the pull request:
## What

Defines a new Feature Flag which allows switching CA trust store.

## Why

Enables gradual roll out of the new CA.

esolitos committed:
feat: Add flag for legacy trust store option

hakonhall commented on flags/src/main/java/com/yahoo/vespa/flags/Flags.java:
```diff
@@ -496,6 +496,13 @@ public class Flags {
+            List.of("marlon"), "2024-12-05", "2025-02-01",
+            "Use legacy trust store for CA, or new one",
+            "Takes effect on restart of OCI containers",
+            SYSTEM);
```
No need for SYSTEM.
The list of dimension here should be the list of dimensions you are explicitly setting before resolving the flag value.

esolitos committed:
refactor(flags): remove unused SYSTEM parameter

Remove the unnecessary SYSTEM parameter in the flag definition for `use-legacy-trust-store` to clean up the code and avoid confusion.
